src/libraries/Microsoft.Extensions.Options/src/OptionsServiceCollectionExtensions.cs

```diff
@@ -26,7 +26,7 @@ public static IServiceCollection AddOptions(this IServiceCollection services)
                 throw new ArgumentNullException(nameof(services));
             }
 
-            services.TryAdd(ServiceDescriptor.Singleton(typeof(IOptions<>), typeof(OptionsManager<>)));
```

Comment:
I really hope people aren't doing this, but I gotta ask: do we know if people get `IOptions<T>`, and then cast/as it either to `OptionsManager<T>` or `IOptionsSnapshot<T>`?

I see `OptionsManager<>` is public, which made me think of this.

Reply:
This is an acceptable breaking change IMO. It's unfortunate that OptionsManager is public but I don't think it's reasonable to tie our hands as a result. This is one of the benefits of using DI (swapping the implementation).

Good find though!

Reply:
I did a query and found some packages using OptionsManager directly but none were casting the default implementation.